tests/src/toCKB_typescript/tests/auction_signertimeout.rs

```diff
@@ -1 +1,243 @@
+use super::ToCKBCellData;
+use crate::toCKB_typescript::utils::{
+    config::*,
+    types::{generated::basic, Error, ToCKBStatus},
+};
+use crate::*;
+use ckb_testtool::{builtin::ALWAYS_SUCCESS, context::Context};
+use ckb_tool::ckb_types::{
+    bytes::Bytes,
+    core::{TransactionBuilder, TransactionView},
+    packed::*,
+    prelude::*,
+};
+use ckb_tool::{ckb_error::assert_error_eq, ckb_script::ScriptError};
+use molecule::prelude::*;
 
+const MAX_CYCLES: u64 = 10_000_000;
+
+#[test]
+fn test_correct_tx_max_time() {
+    let since_max_auction_time = LOCK_TYPE_FLAG | SINCE_TYPE_TIMESTAMP | AUCTION_MAX_TIME;
+    let (context, tx) =
+        build_test_context(375_0000, 375_0000, 0, since_max_auction_time, 2500_0000);
+
+    let cycles = context
+        .verify_tx(&tx, MAX_CYCLES)
+        .expect("pass verification");
+    println!("consume cycles: {}", cycles);
+}
+
+#[test]
+fn test_correct_tx_trigger() {
+    let auction_time = 2 * 24 * 3600;
+    let since = LOCK_TYPE_FLAG | SINCE_TYPE_TIMESTAMP | auction_time;
+    let collateral = 375_0000u64;
+    let bidder_cap = {
+        let init_repayment = collateral * AUCTION_INIT_PERCENT as u64 / 100;
+        init_repayment + (collateral - init_repayment) * auction_time / AUCTION_MAX_TIME
+    };
+
+    let trigger_cap = (collateral - bidder_cap) / 2;
+
+    let (context, tx) = build_test_context(collateral, bidder_cap, trigger_cap, since, 2500_0000);
+
+    let cycles = context
+        .verify_tx(&tx, MAX_CYCLES)
+        .expect("pass verification");
+    println!("consume cycles: {}", cycles);
+}
+
+#[test]
+fn test_wrong_since() {
+    let since = LOCK_TYPE_FLAG | AUCTION_MAX_TIME;
+    let (context, tx) = build_test_context(375_0000, 375_0000, 0, since, 2500_0000);
+
+    let err = context.verify_tx(&tx, MAX_CYCLES).unwrap_err();
+    assert_error_eq!(
+        err,
+        ScriptError::ValidationFailure(Error::InputSinceInvalid as i8)
+    );
+}
+
+#[test]
+fn test_wrong_XT() {
```

Comment:
wrong_XT Error contains 2 kinds:
- wrong XT amount
- wrong XT cell capacity

better to test both